### ISSUE TITLE
Regression tasks: Epistemic and Aleatoric Uncertainty Estimation

### ISSUE THREAD

**Issue by feracero** [IntelLabs/bayesian-torch#22]
Hello, I am trying to use this repository for regression tasks (I can see the examples seem to focus on classification tasks). 

I would like to do estimate epistemic and aleatoric uncertainty for my bayesian neural network as described in 3.1 here https://arxiv.org/pdf/2204.09308.pdf

Could you please provide some guidance on how to obtain the mean and variance used by the final layer to generate the output samples? In this way one could estimate epistemic uncertainty for regression tasks. 

Thank you!

**Comment by famura** [comment]
Hi @feracero, I am currently also thinking about using this repo for regression tasks. Did you have success?

@ranganathkrishnan (and other contributors) would it be possible to add an example? One thing that I am not sure of is how to modify the loss computation which is exemplified for classification in the `Training snippet` section.

**Comment by giulioturrisi** [comment]
@famura, did you manage to perform a regression with this repo? or did you end up with using another one? (if yes, please let me know which :D)

**Comment by famura** [comment]
@giulioturrisi I was putting it off due to other projects. It is still on my plate to try it within the next 1-2 months though. What is your experience?

**Comment by giulioturrisi** [comment]
@famura I have just started now to look around for libraries actually. If i find something nice, I will ping you.

**Comment by ranganathkrishnan** [comment]
> Hi @feracero, I am currently also thinking about using this repo for regression tasks. Did you have success?
> 
> @ranganathkrishnan (and other contributors) would it be possible to add an example? One thing that I am not sure of is how to modify the loss computation which is exemplified for classification in the `Training snippet` section.

Hi @famura It should be straightforward to use model with LinearReparameterization layers with torch.nn.MSELoss() for regression task. I will add an example for regression in the repo.

**Comment by famura** [comment]
Nice, thank you @ranganathkrishnan.

Is there a specific reason why only LSTMs and not GRUs or RNNs are supported [here](https://github.com/IntelLabs/bayesian-torch/blob/main/bayesian_torch/models/dnn_to_bnn.py#L152)? 
Or in other words, why did you have to re-code the LSTM forward pass [here](https://github.com/IntelLabs/bayesian-torch/blob/main/bayesian_torch/layers/variational_layers/rnn_variational.py#L123) instead of using the one from PyTroch?

Update: I think my questions can be answered with "Because we need the KL from the layers that make up the LSTM"

**Comment by ranganathkrishnan** [comment]
> Nice, thank you @ranganathkrishnan.
> 
> Is there a specific reason why only LSTMs and not GRUs or RNNs are supported [here](https://github.com/IntelLabs/bayesian-torch/blob/main/bayesian_torch/models/dnn_to_bnn.py#L152)? Or in other words, why did you have to re-code the LSTM forward pass [here](https://github.com/IntelLabs/bayesian-torch/blob/main/bayesian_torch/layers/variational_layers/rnn_variational.py#L123) instead of using the one from PyTroch?
> 
> Update: I think my questions can be answered with "Because we need the KL from the layers that make up the LSTM"

Hi @famura, No specific reason, we included implementation of reference Bayesian LSTM for time-series prediction tasks, Contributions are welcome through PRs. If you end up implementing Bayesian GRU and RNN layers, please send the pull request. Thanks!

**Comment by staco-tx-mli** [comment]
> > Hi @feracero, I am currently also thinking about using this repo for regression tasks. Did you have success?
> > @ranganathkrishnan (and other contributors) would it be possible to add an example? One thing that I am not sure of is how to modify the loss computation which is exemplified for classification in the `Training snippet` section.
> 
> Hi @famura It should be straightforward to use model with LinearReparameterization layers with torch.nn.MSELoss() for regression task. I will add an example for regression in the repo.

Hello, i am currently also looking into using this library for a regression task. How does one weight the KL divergence compared to the MSE loss? In my case, i have multple outputs where all outputs are standardized (μ = 0, σ = 1) based on the training data.
